File: app/src/main/java/ru/golden/kotlin_weather_station/dto/WeatherDto.java
package ru.golden.kotlin_weather_station.dto;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class WeatherDto implements Serializable {

    @SerializedName("coord")
    private Coord coord;

    @SerializedName("weather")
    private List<Weather> weather = null;

    @SerializedName("base")
    private String base;

    @SerializedName("main")
    private Main main;

    @SerializedName("visibility")
    private Integer visibility;

    @SerializedName("wind")
    private Wind wind;

    @SerializedName("clouds")
    private Clouds clouds;

    @SerializedName("dt")
    private Integer dt;

    @SerializedName("sys")
    private Sys sys;

    @SerializedName("id")
    private Integer id;

    @SerializedName("name")
    private String name;

    @SerializedName("cod")
    private Integer cod;

    private String lastUpdate;
}

File: app/src/main/java/ru/golden/kotlin_weather_station/dto/WeatherDto.kt
package ru.golden.kotlin_weather_station.dto

data class WeatherDto(
	val coord: Coord?,
	val weather: List<Weather>?,
	val base: String?,
	val main: Main?,
	val visibility: Int?,
	val wind: Wind?,
	val clouds: Clouds?,
	val dt: Int?,
	val sys: Sys?,
	val id: Int?,
	val name: String,
	val cod: Int?,
	val lastUpdate: String?
)
